### PR TITLE
fix: improved error handling, avoiding huge log files

### PR DESCRIPTION
**Pull request by ErikBjare**

### CODE REVIEW

File: aw_client/client.py [hunk 1]
```diff
@@ -445,18 +452,44 @@ def should_stop(self) -> bool:
     def _dispatch_request(self) -> None:
         request = self._get_next()
         if not request:
-            self.wait(0.1)  # seconds to wait before re-polling the empty queue
+            self.wait(0.2)  # seconds to wait before re-polling the empty queue
```
Comment (ErikBjare) [inline aw_client/client.py:455]: Made it slightly less agressive.

File: aw_client/client.py [hunk 2]
```diff
@@ -445,18 +452,44 @@ def should_stop(self) -> bool:
...
-                "Failed to send request to aw-server, will queue requests until connection is available."
+                "Connection refused or timeout, will queue requests until connection is available."
             )
+            # wait a bit before retrying, so we don't spam the server (or logs), see:
+            #  - https://github.com/ActivityWatch/activitywatch/issues/815
+            #  - https://github.com/ActivityWatch/activitywatch/issues/756#issuecomment-1266662861
+            sleep(0.5)
             return
+        except req.RequestException as e:
+            if e.response.status_code == 400:
```
Comment (ErikBjare) [inline aw_client/client.py:477]: `e.response` might be None

Comment (ErikBjare) [reply aw_client/client.py]: Made a PR to typeshed to help discover bug like this with mypy: https://github.com/python/typeshed/pull/8989

File: aw_client/client.py [hunk 3]
```diff
@@ -445,18 +452,46 @@ def should_stop(self) -> bool:
...
+            #  - https://github.com/ActivityWatch/activitywatch/issues/815
+            #  - https://github.com/ActivityWatch/activitywatch/issues/756#issuecomment-1266662861
+            sleep(0.5)
             return
+        except req.RequestException as e:
+            if e.response and e.response.status_code == 400:
+                # HTTP 400 - Bad request
+                # Example case: https://github.com/ActivityWatch/activitywatch/issues/815
+                # We don't want to retry, because a bad payload is likely to fail forever.
+                logger.error("Bad request, not retrying: {}".format(request.data))
```
Comment (johan-bjareholt) [inline aw_client/client.py:481]: Maybe also log the response from the server?

**Pull request merged**